### PR TITLE
feat(form-textarea): add `noAutoShrink` prop (closes #2664)

### LINKED ISSUE #2664: Form textarea with control sizing auto collapses on input
Just try it out directly in the demo page:
https://bootstrap-vue.js.org/docs/components/form-textarea/

go to the section "Control sizing" and grab the bottom right handle to make the text area bigger.

As soon as I paste some text into it, it collapses to the default size, which I do not believe is what would be expected.

### PR DESCRIPTION
**Pull request by jackmu95**
## Description of Pull Request:

Currently the `<b-form-textarea>` resizes to min-height (or the content height, whichever is greater) even when it had a taller height before.
~This PR fixes this misbehavior.~

Add in a new prop that will allow users to disable the auto shrink to content height (i.e. making max height "sticky").

Closes #2664.

To Do:
- [ ] docs update

## PR checklist:

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [x] Enhancement to an existing feature
- [ ] ARIA accessibility
- [ ] Documentation update
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e.
      `fixes #xxxx[,#xxxx]`, where "xxxx" is the issue number)
- [x] The PR should address only one issue or feature. If adding multiple features or fixing a bug
      and adding a new feature, break them into separate PRs if at all possible.
- [x] PR titles should following the
      [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e.
      "fix(alert): not alerting during SSR render", "docs(badge): Updated pill examples, fix typos",
      "chore: fix typo in docs", etc). **This is very important, as the `CHANGELOG` is generated
      from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [x] Includes documentation updates (including updating the component's `package.json` for slot and
      event changes)
- [ ] New/updated tests are included and passing (if required)
- [x] Existing test suites are passing
- [x] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (does it affect screen reader users or
      keyboard only users? clickable items should be in the tab index, etc)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a
      suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: src/components/form-textarea/form-textarea.js [hunk 1]
```diff
@@ -98,14 +102,19 @@ export default {
         (parseFloat(computedStyle.paddingBottom) || 0)
       // Calculate content height in "rows"
       const contentRows = (el.scrollHeight - offset) / lineHeight
-      // Put the old height back (needed when new height is equal to old height!)
-      el.style.height = oldHeight
```
Comment (tmorehouse) [inline src/components/form-textarea/form-textarea.js]: This line is needed to handle certain edge cases when changing the height.

Comment (tmorehouse) [reply src/components/form-textarea/form-textarea.js]: Note that this is altering the element directly, not returning a value.

Comment (tmorehouse) [reply src/components/form-textarea/form-textarea.js]: And this also triggers a reflow on the element.  Can't be placed inside an "if".

**Comment by tmorehouse** [comment]
Please note that original behaviour is the intended behaviour and **is not a bug**.

**Comment by tmorehouse** [comment]
Adding in an extra check to handle the case when no max-rows is given and manual dragged resize is enabled.

**Comment by jackmu95** [comment]
@tmorehouse I've added some docs and for me the auto-height handling is completely broken. Can you confirm that in the deployment preview?

**Comment by tmorehouse** [comment]
Ok will check here shortly.

**Comment by tmorehouse** [comment]
@jackmu95 you were missing the `max-rows` props on the auto-height examples.  I've added them in and the examples work now.

**Comment by jackmu95** [comment]
@tmorehouse This removes the possibility to resize the textarea at all for me.

OSX 10.14.3
Chrome 72

**Comment by tmorehouse** [comment]
@jackmu95 that is correct (as was/is documented).  Using the auto-resize feature (which is triggered by setting `max-rows`), the resize handle is removed from the `textarea` via CSS.

**Pull request merged**